piecrust/src/store.rs

```diff
@@ -430,7 +431,12 @@ fn write_commit_inner<P: AsRef<Path>>(
             bytecode_dir,
             memory_dir,
             base: base.map(|inner| {
-                let base_root = inner.index.root();
+                let base_root_ref = inner.index.root();
+                let base_root = *base_root_ref;
+
+                // We manually drop the reference to the root hash to allow us
+                // to move out of `inner` when we return.
+                drop(base_root_ref);
```

Comment:
Isn't `let base_root = *inner.index.root();` enough to release the ref to base_root?

Reply:
You're right, fixed.